internal/backend/remote-state/http/client.go

```diff
@@ -61,6 +61,7 @@ func (c *httpClient) httpRequest(method string, url *url.URL, data *[]byte, what
 	// Work with data/body
 	if data != nil {
 		req.Header.Set("Content-Type", "application/json")
+		req.Header.Set("Opentofu-http-client/1.1")
```

Comment:
I would prefer if we could use existing mechanisms to create a user agent.

For reference, you can see this code here: https://github.com/opentofu/opentofu/blob/cb2e9119aa75eeb8e1fa175e2b7a205a4fef129f/internal/httpclient/useragent.go#L35-L51

This is used throughout the codebase to set the useragent on clients.